Client/Frontend/BraveVPN/BraveVPN.swift

```diff
@@ -535,4 +560,27 @@ class BraveVPN {
         GRDVPNHelper.saveAll(inOneBoxHostname: hostname)
         GRDGatewayAPI.shared().apiHostname = hostname
     }
+    
+    // MARK: - Server selection
+    
+    /// Returns a list of available vpn regions to switch to.
+    static func requestAllServerRegions(_ completion: @escaping ([VPNRegion]?) -> Void) {
+        housekeepingApi.requestAllServerRegions { regionList, success in
+            if !success {
+                logAndStoreError("requestAllServerRegions call failed")
+                completion(nil)
+                return
+            }
+            
+            guard let regionList = regionList,
+                  let data = try? JSONSerialization.data(withJSONObject: regionList,
+                                                         options: .fragmentsAllowed) else {
+                logAndStoreError("failed to deserialize server regions data")
+                completion(nil)
+                return
+            }
+            
+            completion(try? JSONDecoder().decode([VPNRegion].self, from: data))
```

Comment:
May want to add another `logAndStoreError` based on `JSONDecoder`'s potential parsing errors